pkg/resources/management.cattle.io/v3/project/validator_test.go

```diff
@@ -0,0 +1,698 @@
+			operation: admissionv1.Update,
+			oldProject: &v3.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "testcluster",
+				},
+				Spec: v3.ProjectSpec{
+					ResourceQuota: &v3.ProjectResourceQuota{
+						Limit: v3.ResourceQuotaLimit{
+							ConfigMaps: "100",
+						},
+						UsedLimit: v3.ResourceQuotaLimit{
+							ConfigMaps: "100",
+						},
+					},
+					NamespaceDefaultResourceQuota: &v3.NamespaceResourceQuota{
+						Limit: v3.ResourceQuotaLimit{
+							ConfigMaps: "50",
+						},
+					},
+				},
+			},
+			newProject: &v3.Project{
+				Spec: v3.ProjectSpec{
+					ResourceQuota: &v3.ProjectResourceQuota{
+						Limit: v3.ResourceQuotaLimit{
+							ConfigMaps: "60",
+						},
+						UsedLimit: v3.ResourceQuotaLimit{
+							ConfigMaps: "100",
+						},
+					},
+					NamespaceDefaultResourceQuota: &v3.NamespaceResourceQuota{
+						Limit: v3.ResourceQuotaLimit{
+							ConfigMaps: "50",
+						},
+					},
+				},
+			},
+			wantAllowed: false,
+		},
+		{
+			name:      "update with project quota less than namespace quota times number of namespaces",
+			operation: admissionv1.Update,
+			oldProject: &v3.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "testcluster",
+				},
+				Spec: v3.ProjectSpec{
+					ResourceQuota: &v3.ProjectResourceQuota{
+						Limit: v3.ResourceQuotaLimit{
+							ConfigMaps: "20",
+						},
+						UsedLimit: v3.ResourceQuotaLimit{
+							ConfigMaps: "10",
+						},
+					},
+					NamespaceDefaultResourceQuota: &v3.NamespaceResourceQuota{
+						Limit: v3.ResourceQuotaLimit{
+							ConfigMaps: "5",
+						},
+					},
+				},
+			},
+			newProject: &v3.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "testcluster",
+				},
+				Spec: v3.ProjectSpec{
+					ResourceQuota: &v3.ProjectResourceQuota{
+						Limit: v3.ResourceQuotaLimit{
+							ConfigMaps: "20",
+							LimitsCPU:  "15m",
+						},
+						UsedLimit: v3.ResourceQuotaLimit{
+							ConfigMaps: "10",
+						},
+					},
+					NamespaceDefaultResourceQuota: &v3.NamespaceResourceQuota{
+						Limit: v3.ResourceQuotaLimit{
+							ConfigMaps: "5",
+							LimitsCPU:  "10m",
+						},
+					},
+				},
+			},
+			namespaces: []*corev1.Namespace{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "testns1",
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "testns2",
+					},
+				},
+			},
+			wantAllowed: false,
+		},
+		{
+			name:      "update with fields changed in project quota less than used quota",
+			operation: admissionv1.Update,
+			oldProject: &v3.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "testcluster",
+				},
+				Spec: v3.ProjectSpec{
+					ResourceQuota: &v3.ProjectResourceQuota{
+						Limit: v3.ResourceQuotaLimit{
+							LimitsCPU: "100m",
+						},
+						UsedLimit: v3.ResourceQuotaLimit{
+							ConfigMaps: "100",
```

Comment:
I would expect this case to use the same resource in this field, was this use of configmaps here intentional?